File: java/src/main/java/webmis/util/Os.java
package webmis.util;

/* 设备信息 */
public class Os {

  /* OS */
  public static String System(String user_agent) {
    if(user_agent.contains("Win64") || user_agent.contains("Windows NT"))
      return "Windows";
    else if(user_agent.contains("Linux"))
      return "Linux";
    else if(user_agent.contains("Mac OS"))
      return "MacOS";
    else if(user_agent.contains("Unix"))
      return "Unix";
    else
      return "Other";
  }

  /* Browser */
  public static String Browser(String user_agent) {
    if(user_agent.contains("MSIE"))
      return "IE";
    else if(user_agent.contains("Netscape"))
      return "Netscape";
    else if(user_agent.contains("Opera"))
      return "Opera";
    else if(user_agent.contains("Firefox"))
      return "Firefox";
    else if(user_agent.contains("Chrome"))
      return "Chrome";
    else if(user_agent.contains("Safari"))
      return "Safari";
    else
      return "Other";
  }
  
}

File: python/util/os.py
# 设备信息
class Os:

  # OS
  def System(user_agent: str):
    if 'Win64' in user_agent or 'Windows NT' in user_agent :
      return 'Windows'
    elif 'Linux' in user_agent :
      return 'Linux'
    elif 'Mac' in user_agent :
      return 'Mac OS'
    elif 'Unix' in user_agent :
      return 'Unix'
    else :
      return 'Other'

  # Browser
  def Browser(user_agent: str):
    if 'MSIE' in user_agent :
      return 'IE'
    elif 'Netscape' in user_agent :
      return 'Netscape'
    elif 'Opera' in user_agent :
      return 'Opera'
    elif 'Firefox' in user_agent :
      return 'Firefox'
    elif 'Chrome' in user_agent :
      return 'Chrome'
    elif 'Safari' in user_agent :
      return 'Safari'
    else :
      return 'Other'